### PR TITLE
109 dynamic data

### PR DESCRIPTION
**Pull request by cchauche**
Sorry its another big one but hopefully form here on out I can do much smaller targeted commits

Mostly its just all the work necessary to get the right dynamic data in the correct format to correct module.

I did make a modification to App.jsx but it was only the props being passed down to the reviewWidget.

### CODE REVIEW

File: client/components/ReviewsWidget.jsx [hunk 1]
```diff
@@ -1,20 +1,167 @@
...
     return (
       <div className="reviews-widget">
         <h3>{'Ratings & Reviews'}</h3>
         <div className="reviews-container">
-          <RatingBreakdown />
-          <ReviewList />
+          <RatingBreakdown
+            reviewMetaData={this.props.reviewMetaData}
+            reviewAverage={this.props.reviewAverage}
+            reviewCount={this.state.reviewCount}
```
Comment (zrealberg) [inline client/components/ReviewsWidget.jsx]: this prop (reviewCount) is undefined further down the tree (RatingsBreakdown, ReviewsSummary, etc). I think this value should be `this.props.reviewCount` not `.state.reviewCount`.

File: client/components/reviewWidget/RatingBreakdown.jsx [hunk 2]
```diff
@@ -3,12 +3,20 @@ import ReviewsSummary from './breakdownComponents/ReviewsSummary.jsx';
 import ReviewsBreakdown from './breakdownComponents/ReviewsBreakdown.jsx';
 import CharacteristicsBreakdown from './breakdownComponents/CharacteristicsBreakdown.jsx';
 
-const RatingBreakdown = () => {
+const RatingBreakdown = ({reviewAverage, reviewMetaData, reviewCount}) => {
```
Comment (zrealberg) [inline client/components/reviewWidget/RatingBreakdown.jsx:6]: reviewCount is undefined, see the note I left in components/ReviewsWidget.jsx

File: client/components/reviewWidget/ReviewCard.jsx [hunk 3]
```diff
@@ -1,36 +1,60 @@
...
 
-const ReviewCard = () => {
+const ReviewCard = ({ review, index }) => {
+  // Conditionally render a response if present
+  let response = null;
+  if (review.response) {
+    response = (
+      <div className="review-response">
+        <h6>Response:</h6>
+        <p>{review.response}</p>
```
Comment (zrealberg) [inline client/components/reviewWidget/ReviewCard.jsx:11]: I think this be {review.body}. Same for line 7. I don't see these elements on the page, probably for this reason.

File: client/components/reviewWidget/breakdownComponents/ReviewsBreakdown.jsx [hunk 4]
```diff
@@ -1,16 +1,21 @@
...
-      <ReviewBar />
-      <ReviewBar />
-    </section>
+const ReviewsBreakdown = ({ ratings }) => {
+  let bars = [];
+  let totalRatings = Object.values(ratings).reduce(
+    (sum, num) => sum + parseInt(num),
+    0
   );
+  for (let i = 0; i < 6; i++) {
```
Comment (zrealberg) [inline client/components/reviewWidget/breakdownComponents/ReviewsBreakdown.jsx:10]: pretty nice code here!

File: client/components/reviewWidget/breakdownComponents/ReviewsSummary.jsx [hunk 5]
```diff
@@ -1,16 +1,28 @@
...
 import StarAverage from '../../shared/StarAverage.jsx';
 
-const ReviewsSummary = () => {
+const calcPercentRecommended = (recommended) => {
+  let recomend = parseInt(recommended.true);
+  let total = recomend + parseInt(recommended.false);
+  return recomend / total * 100 + '%';
+};
+
+const ReviewsSummary = ({reviewAverage, recommended, reviewCount}) => {
```
Comment (zrealberg) [inline client/components/reviewWidget/breakdownComponents/ReviewsSummary.jsx:10]: chrome devtools says reviewCount is undefined, probably related to issue I found earlier.

**Review by zrealberg — COMMENTED** [review]
So much fun to dive into someone else work ahhaha. 

I see that a lot of work is still in progress, but it looks good for the most part. 

There's one issue with props being passed down that I flagged a few times. 

Your inclusion of classes is well done, I haven't done that. Also some very fancy ES6, chaining and other code in there.

**Review by zrealberg — APPROVED** [review]
So much fun to dive into someone else work ahhaha.

I see that a lot of work is still in progress, but it looks good for the most part.

There's one issue with props being passed down that I flagged a few times.

Your inclusion of classes is well done, I haven't done that. Also some very fancy ES6, chaining and other code in there.

File: client/components/ReviewsWidget.jsx (hunk 1 above)
Comment (cchauche) [reply client/components/ReviewsWidget.jsx]: yup, thank you!

Comment (cchauche) [reply client/components/ReviewsWidget.jsx]: @zrealberg fixed and thank you again!

File: client/components/ReviewsWidget.jsx [hunk 6]
```diff
@@ -1,20 +1,167 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      filteredReviews: dummyData,
+      productReviews: dummyData,
+      page: 1,
+      numToDisplay: 2,
+      sortOrder: 'relevant',
+    };
+  }
```
Comment (jimmyko360) [inline client/components/ReviewsWidget.jsx:136]: Looks like productReviews and filteredReviews are the same for now. Were you planning on these holding different pieces of state in the future?

**Review by jimmyko360 — APPROVED** [review]
Looks great! All of the data manipulation with the stars, date, etc. is great. The HTML structure is super organized too.

**Comment by jimmyko360** [comment]
@cchauche the edit you made to App.jsx where you pass props to your widget is creating a conflict. I'm not sure why, as it should be easy for Github to swap out those lines in theory. That should be the last bit before we merge this into the master branch though.

File: client/components/ReviewsWidget.jsx (hunk 6 above)
Comment (cchauche) [reply client/components/ReviewsWidget.jsx:136]: Yeah it is a work in progress I haven't quite figure out yet how these relate or if I need both so for now it just gets dummy data.  I think the approach I'll take is that productReviews will hold all the reviews that have been downloaded and then the filtering script will take productReviews apply the filter and update filteredReviews which then gets passed down.  Just my working concept definitely looking for any clever insights people might have

Comment (jimmyko360) [reply client/components/ReviewsWidget.jsx:136]: Oh that's actually a really clever way to implement that!

**Pull request merged**